pkg/virt-controller/watch/vmi.go

```diff
@@ -1811,6 +1825,25 @@ func (c *VMIController) waitForFirstConsumerTemporaryPods(vmi *virtv1.VirtualMac
 	return temporaryPods, nil
 }
 
+func (c *VMIController) MarkMemoryHotplugReady(vmi *virtv1.VirtualMachineInstance) {
+	if vmi.Labels == nil {
+		vmi.Labels = map[string]string{}
+	}
+	vmi.Labels[virtv1.MemoryHotplugReadyLabel] = ""
+}
+
+func (c *VMIController) isMemoryHotplugReady(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) bool {
+	expectedPodMemReq := vmi.Spec.Domain.Memory.Guest.DeepCopy()
+	expectedPodMemReq.Add(services.GetMemoryOverhead(vmi, c.clusterConfig.GetClusterCPUArch(), c.clusterConfig.GetConfig().AdditionalGuestMemoryOverheadRatio))
```

Comment:
how do we know that `AdditionalGuestMemoryOverheadRatio` didn't change between the time that the pod was created and now()?

Reply:
also, do we allow memory hot plug in cases when the guest memory is different from requests.memory?

Reply:
Yes we do, do you think we shouldn't?

Reply:
About your first question, that's a good one. We don't if someone does that the hotplug would just get stuck. What if we don't allow hotplug if `AdditionalGuestMemoryOverheadRatio` is set? too strict?

Reply:
I think we can reach out to the same inconsistent scenario also w/o mem. hotplug. Consider a case where you launch a VM, the VMI pod is pending creation due to insufficient resources (or due to any other reason), and during that time you've changed `AdditionalGuestMemoryOverheadRatio`, so the VMI pod will be created with the outdated ratio.

Reply:
Oh I see what you mean now. A situation where virt-controller and virt-handler read different values from the KV CR because the user had changed the value in between

Reply:
Maybe we can forbid the user from using `vm.spec.template.spec.domain.resources.requests.memory` and capture the calculated desired memory request in the VMI, read this value in `isMemoryHotplugReady` and compare to compute pod

Reply:
I fixed it by storing the `AdditionalGuestMemoryOverheadRatio` (using a label) when the hotplug gets triggered so that if someone changes it on the fly it doesn't affect hotplug.